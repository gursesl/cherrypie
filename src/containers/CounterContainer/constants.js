export const INCREMENT_ACTION = 'app/CounterContainer/INCREMENT_ACTION'
export const DECREMENT_ACTION = 'app/CounterContainer/ACTION_DECREMENT'
export const INCREMENT_ACTION_ASYNC = 'app/CounterContainer/INCREMENT_ACTION_ASYNC'
