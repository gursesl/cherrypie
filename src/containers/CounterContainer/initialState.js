import { fromJS } from 'immutable'

const initialState = fromJS({
  value: 0,
})

export default initialState
