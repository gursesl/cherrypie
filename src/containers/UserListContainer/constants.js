export const USERS_FETCH_START = 'app/UserListContainer/USERS_FETCH_START'
export const USERS_FETCH_SUCCESS = 'app/UserListContainer/USERS_FETCH_SUCCESS'
export const USERS_FETCH_FAILURE = 'app/UserListContainer/USERS_FETCH_FAILURE'
