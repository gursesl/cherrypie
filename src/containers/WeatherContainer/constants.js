export const WEATHER_DATA_FETCH_START = 'app/WeatherContainer/WEATHER_DATA_FETCH_START'
export const WEATHER_DATA_FETCH_SUCCESS = 'app/WeatherContainer/WEATHER_DATA_FETCH_SUCCESS'
export const WEATHER_DATA_FETCH_FAILURE = 'app/WeatherContainer/WEATHER_DATA_FETCH_FAILURE'
export const SELECTOR_WEATHER = 'app/WeatherContainer/SELECTOR_WEATHER'
export const SELECTOR_WEATHER_ZIP = 'app/WeatherContainer/SELECTOR_WEATHER_ZIP'
export const SELECTOR_WEATHER_CITY = 'app/WeatherContainer/SELECTOR_WEATHER_CITY'
export const SELECTOR_WEATHER_RESULTS = 'app/WeatherContainer/SELECTOR_WEATHER_RESULTS'
export const SELECTOR_WEATHER_ERROR = 'app/WeatherContainer/SELECTOR_WEATHER_ERROR'
