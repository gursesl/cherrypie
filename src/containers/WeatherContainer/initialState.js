import { fromJS } from 'immutable'
import {
  SELECTOR_WEATHER_ZIP,
  SELECTOR_WEATHER_CITY,
  SELECTOR_WEATHER_RESULTS,
  SELECTOR_WEATHER_ERROR,
} from './constants'

const initialState = fromJS({
  [SELECTOR_WEATHER_ZIP]: '20165',
  [SELECTOR_WEATHER_CITY]: 'Sterling',
  [SELECTOR_WEATHER_RESULTS]: [],
  [SELECTOR_WEATHER_ERROR]: 'Load error',
})

export default initialState
