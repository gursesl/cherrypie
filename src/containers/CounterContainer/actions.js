import * as c from './constants'

export function incrementAction() {
  return {
    type: c.INCREMENT_ACTION,
  }
}

export function decrementAction() {
  return {
    type: c.DECREMENT_ACTION,
  }
}

export function incrementActionAsync() {
  return {
    type: c.INCREMENT_ACTION_ASYNC,
  }
}
