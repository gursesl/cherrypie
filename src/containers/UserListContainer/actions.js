import * as c from './constants'

export function usersFetchStart() {
  return { type: c.USERS_FETCH_START }
}

export function usersFetchSuccess(users) {
  return {
    type: c.USERS_FETCH_SUCCESS,
    payload: users,
  }
}

export function usersFetchFailure(err) {
  return {
    type: c.USERS_FETCH_FAILURE,
    error: err,
  }
}
