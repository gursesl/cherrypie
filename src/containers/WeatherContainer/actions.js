import * as c from './constants'

export function weatherDataFetchStart() {
  return { type: c.WEATHER_DATA_FETCH_START }
}

export function weatherDataFetchSuccess(cities) {
  return {
    type: c.WEATHER_DATA_FETCH_SUCCESS,
    payload: cities,
  }
}

export function weatherDataFetchFailure(err) {
  return {
    type: c.WEATHER_DATA_FETCH_FAILURE,
    error: err,
  }
}
