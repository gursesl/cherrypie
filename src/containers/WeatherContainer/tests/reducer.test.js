import { fromJS } from 'immutable'
import initialState from '../initialState'
import weatherContainerReducer, { processResultData } from '../reducer'
import * as a from '../actions'
import * as c from '../constants'
import * as m from './mockdata'

const state = initialState

describe('WeatherContainer:reducer', () => {
  it('should return the initial state', () => {
    expect(weatherContainerReducer(undefined, {})).toEqual(initialState)
  })

  it('should handle weatherDataFetchStart action correctly', () => {
    expect(weatherContainerReducer(state, a.weatherDataFetchStart())).toEqual(state)
  })

  it('should handle search by zip code', () => {
    expect(weatherContainerReducer(
      state.set(c.SELECTOR_WEATHER_ZIP, ''),
      a.weatherDataFetchStart('20002')
    )).toEqual(state.set(c.SELECTOR_WEATHER_ZIP, '20002'))
  })

  it('should handle search by empty zip code', () => {
    expect(weatherContainerReducer(
      state.set(c.SELECTOR_WEATHER_ZIP, ''),
      a.weatherDataFetchStart('')
    )).toEqual(state.set(c.SELECTOR_WEATHER_ZIP, ''))
  })

  it('should handle search by city name', () => {
    expect(weatherContainerReducer(
      state.set(c.SELECTOR_WEATHER_CITY, ''),
      a.weatherDataFetchStart('Washington, DC')
    )).toEqual(state.set(c.SELECTOR_WEATHER_CITY, 'Washington, DC'))
  })

  it('should handle search by empty city name', () => {
    expect(weatherContainerReducer(
      state.set(c.SELECTOR_WEATHER_CITY, ''),
      a.weatherDataFetchStart('')
    )).toEqual(state.set(c.SELECTOR_WEATHER_CITY, ''))
  })

  it('should handle weatherDataFetchSuccess action correctly with a payload', () => {
    const expectedState = state.set(c.SELECTOR_WEATHER_RESULTS, processResultData(m.mockData.toJS()))
    expect(weatherContainerReducer(
      state,
      a.weatherDataFetchSuccess(m.mockData.toJS())
    )).toEqual(expectedState)
  })

  it('should handle weatherDataFetchSuccess action correctly w/o a payload', () => {
    const expectedState = state.set(c.SELECTOR_WEATHER_RESULTS, fromJS([]))
    expect(weatherContainerReducer(state, a.weatherDataFetchSuccess())).toEqual(expectedState)
  })

  it('should handle weatherDataFetchFailure action correctly with a load', () => {
    const expectedState = state.set(c.SELECTOR_WEATHER_ERROR, m.mockError)
    expect(weatherContainerReducer(
      state,
      a.weatherDataFetchFailure(m.mockError)
    )).toEqual(expectedState)
  })

  it('should handle weatherDataFetchFailure action correctly w/o a load', () => {
    const expectedState = state.set(c.SELECTOR_WEATHER_ERROR, '')
    expect(weatherContainerReducer(state, a.weatherDataFetchFailure(''))).toEqual(expectedState)
  })

  // Data processor test
  it('should process result data correctly', () => {
    expect(processResultData(m.mockData.toJS())).toEqual(m.processedMockData)
  })

  // Snapshot tests
  it('handles the weatherDataFetchStart action snapshot', () => {
    expect(weatherContainerReducer(state, a.weatherDataFetchStart(fromJS({
      type: c.WEATHER_DATA_FETCH_START,
      payload: 'Sterling',
    })))).toMatchSnapshot()
  })

  it('handles the weatherDataFetchSuccess action snapshot', () => {
    expect(weatherContainerReducer(state, a.weatherDataFetchSuccess(m.mockData.toJS()))).toMatchSnapshot()
  })

  it('handles the weatherDataFetchFailure action snapshot', () => {
    expect(weatherContainerReducer(state, a.weatherDataFetchFailure(m.mockError))).toMatchSnapshot()
  })
})
