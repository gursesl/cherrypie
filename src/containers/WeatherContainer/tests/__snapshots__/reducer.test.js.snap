// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherContainer:reducer handles the weatherDataFetchFailure action snapshot 1`] = `
Immutable.Map {
  "app/WeatherContainer/SELECTOR_WEATHER_ZIP": "20165",
  "app/WeatherContainer/SELECTOR_WEATHER_CITY": "Sterling",
  "app/WeatherContainer/SELECTOR_WEATHER_RESULTS": Immutable.Map {},
  "app/WeatherContainer/SELECTOR_WEATHER_ERROR": Immutable.Map {
    "type": 1,
    "message": "An error occurred.",
  },
}
`;

exports[`WeatherContainer:reducer handles the weatherDataFetchStart action snapshot 1`] = `Object {}`;

exports[`WeatherContainer:reducer handles the weatherDataFetchSuccess action snapshot 1`] = `
Immutable.Map {
  "app/WeatherContainer/SELECTOR_WEATHER_ZIP": "20165",
  "app/WeatherContainer/SELECTOR_WEATHER_CITY": "Sterling",
  "app/WeatherContainer/SELECTOR_WEATHER_RESULTS": Immutable.Map {},
  "app/WeatherContainer/SELECTOR_WEATHER_ERROR": "Load error",
}
`;
